scripts/mgear/maya/shifter/__init__.py

```diff
@@ -1,76 +1,38 @@
-# MGEAR is under the terms of the MIT License
-
-# Copyright (c) 2016 Jeremie Passerin, Miquel Campos
-
-# Permission is hereby granted, free of charge, to any person obtaining a copy
-# of this software and associated documentation files (the "Software"), to deal
-# in the Software without restriction, including without limitation the rights
-# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
-# copies of the Software, and to permit persons to whom the Software is
-# furnished to do so, subject to the following conditions:
-
-# The above copyright notice and this permission notice shall be included in all
-# copies or substantial portions of the Software.
-
-# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
-# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
-# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
-# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
-# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
-# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
-# SOFTWARE.
-
-# Author:     Jeremie Passerin      geerem@hotmail.com  www.jeremiepasserin.com
-# Author:     Miquel Campos         hello@miquel-campos.com  www.miquel-campos.com
-# Date:       2016 / 10 / 10
-
-"""
-Shifter base rig class.
-"""
-
-
-#############################################
-# GLOBAL
-#############################################
-# Built in
 import os.path
 import datetime
 import getpass
 
 # Maya
 import pymel.core as pm
-import pymel.core.datatypes as dt
+from pymel.core import datatypes
 from pymel import versions
 
 # mgear
 import mgear
 import mgear.maya.utils
-from mgear.maya.shifter.guide import RigGuide
-from mgear.maya.shifter.guide import helperSlots
+from mgear.maya.shifter.guide import RigGuide, helperSlots
```

Comment:
These can be reduced to..

```python
from .guide import RigGuide
```

Though I'd suggest we keep the namespace, and make it..

```python
from . import guide
```

Reply:
something like this?
'''
from . import guide
from guide import RigGuide, helperSlots
'''

Reply:
I was thinking more along these lines..

```python
from . import guide
...
self.guide = guide.RigGuide()
```

If you stick with namespaces, you can then also reduce variable names. For example..

```python
self.guide = guide.Rig()
```

See how it reduces the duplicate mention of "guide"?

Reply:
Not sure if I understand this correctly.  Do you mind to change this file for a full example? Just to be sure.  Thanks!

Reply:
Sure, no problem. I'll take a look at this tomorrow morning.

Reply:
Ok, I've pushed an example.

1. The namespace of the `component` and `guide` modules are maintained
2. Names are shortened to take into consideration the namespace
3. Backwards compatibility is maintained via aliases

This highlights an issue with the current names of the modules, `guide` is a module but also an often used variable name, as is `component`. I can't say there's a straightforward solution, except to avoid nouns for module names, and reserve those for variables. For example `utility.py` and `io.py` is unlikely to conflict with variables, whereas `object` and `item` likely are.

The shortening isn't as important as the overall PEP8 changes, so we could delay those until we're done with that.

Reply:
Thank you,  I am checking now the changes

Reply:
Oh I get it now! 
So we use simple names but always inside the namespace. So is easily identifiable from where is coming.
While creating the alias just in case for back compatibility. 
looks good for me. thanks!

Reply:
Cool, glad you like it. :)

Once we get to defining an API for mgear, this will become even crisper and easily readable. I'll pop open an issue with some thoughts about that later today.